src/ASM/ASMs2D.C

```diff
@@ -590,19 +590,23 @@ bool ASMs2D::assignNodeNumbers (BlockNodes& nodes, int basis)
 }
 
 
-bool ASMs2D::connectPatch (int edge, ASMs2D& neighbor, int nedge,
+bool ASMs2D::connectPatch (int edge, ASM2D& neighbor, int nedge,
                            bool revers, int, bool coordCheck, int thick)
 {
+  ASMs2D* sn = dynamic_cast<ASMs2D*>(&neighbor);
+  if (!sn)
+    return false;
+
```

Comment:
Could consider a little bit longer variable name here, e.g., `neighS` (also since we use `neighMx` in the mixed case)